R/exercise.R

```diff
@@ -463,41 +468,130 @@ exercise_code_chunks <- function(exercise) {
 }
 
 
-exercise_result_timeout <- function() {
+exercise_result_timeout <- function(exercise) {
   exercise_result_error(
     "Error: Your code ran longer than the permitted timelimit for this exercise.",
-    timeout_exceeded = TRUE
+    timeout_exceeded = TRUE,
+    exercise = exercise
   )
 }
 
 # @param timeout_exceeded represents whether or not the error was triggered
 #   because the exercise exceeded the timeout. Use NA if unknown
-exercise_result_error <- function(error_message, feedback = NULL, timeout_exceeded = NA) {
+exercise_result_error <- function(
+  error_message,
+  feedback = NULL,
+  timeout_exceeded = NA,
+  exercise = NULL
+) {
   exercise_result(
     feedback = feedback,
     timeout_exceeded = timeout_exceeded,
     error_message = error_message,
-    html_output = error_message_html(error_message)
+    html_output = error_message_html(error_message, exercise),
+    exercise = exercise
   )
 }
 
-exercise_result <- function(feedback = NULL, html_output = NULL,
-                            error_message = NULL, timeout_exceeded = FALSE) {
+exercise_result <- function(
+  feedback = NULL,
+  html_output = NULL,
+  error_message = NULL,
+  timeout_exceeded = FALSE,
+  exercise = NULL
+) {
+
+  # When `exercise` is empty, we return a list of NULL
+  # `exercise` is NULL when there was no code sent (i.e the
+  # student hasn't entered any code)
+  if (is.null(exercise)){
+    return(
+      structure(
+        list(
+          feedback = feedback,
+          error_message = error_message,
+          timeout_exceeded = timeout_exceeded,
+          html_output = html_output
+        ),
+        class = "learnr_exercise_result"
+      )
+    )
+  }
+
   feedback <- feedback_validated(feedback)
-  feedback_html <- feedback_as_html(feedback)
+  feedback_html <- feedback_as_html(feedback, exercise)
+
+  if (
+    is.null(exercise$check) &&
+    is.null(exercise$code_check)
+  ){
+    exercise.submitted_feedback <- TRUE
```

Comment:
When "run"ning code, should this value be set to `FALSE`? (So that there is no feedback)

```suggestion
    exercise.submitted_feedback <- FALSE
```

Reply:
When clicking Run Code there is no feedback shown (there used to be a message but we decided to remove it), so as far as I can tell, it doesn't need to be switched here.